src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala

```diff
@@ -0,0 +1,83 @@
+package org.broadinstitute.dsde.firecloud.service
+
+import org.broadinstitute.dsde.firecloud.FireCloudConfig
+import org.broadinstitute.dsde.firecloud.dataaccess.HttpGoogleServicesDAO
+import org.broadinstitute.dsde.firecloud.model.OAuthException
+import org.slf4j.LoggerFactory
+import spray.http.Uri._
+import spray.http.{StatusCodes, Uri}
+import spray.routing._
+
+// see https://developers.google.com/identity/protocols/OAuth2WebServer
+
+trait OAuthService extends HttpService with PerRequestCreator with FireCloudDirectives {
+
+  private implicit val executionContext = actorRefFactory.dispatcher
+  lazy val log = LoggerFactory.getLogger(getClass)
+
+  val routes: Route =
+    path("login") {
+      get {
+        parameter("path") { userpath =>
+          // TODO: future story: generate and persist unique token along with the user path
+          val state = userpath
+
+          // approval prompt - "force" to force a refresh token
+          // TODO: future story: make dynamic based on whether or not the user has a refresh token stored already
+          val approvalPrompt = "auto"
+
+          try {
+            // create the authentication url and redirect the browser
+            val gcsAuthUrl = HttpGoogleServicesDAO.getGoogleRedirectURI(state, approvalPrompt)
+            redirect(gcsAuthUrl, StatusCodes.TemporaryRedirect)
+          } catch {
+            /* we don't expect any exceptions here; if we get any, it's likely
+                a misconfiguration of our client id/secrets/callback. But since this is about
+                authentication, we are extra careful.
+             */
+            case e: Exception => {
+              log.error("problem during OAuth redirect", e)
+              // TODO: here and elsewhere, redirect to a UI error page instead of issuing a 401?
+              complete(StatusCodes.Unauthorized, e.getMessage)
```

Comment:
I would suggest removing the TODO and adding a story for this if/when we actually run into this issue.